tests/metrics/test_scoring.py

```diff
@@ -166,3 +170,52 @@ def test_raises(self, metric: MetricAtK, arg_names: tp.List[str]) -> None:
         kwargs = {name: getattr(self, name) for name in arg_names}
         with pytest.raises(ValueError):
             calc_metrics({"m": metric}, **kwargs)
+
+    def test_success_debias(self) -> None:
+        debias_config = DebiasConfig(iqr_coef=1.5, random_state=32)
+        debias_metrics = {
+            "debias_precision@3": Precision(k=3, debias_config=debias_config),
+            "debias_rprecision@3": Precision(k=3, r_precision=True, debias_config=debias_config),
+            "debias_recall@3": Recall(k=3, debias_config=debias_config),
+            "debias_f1beta@3": F1Beta(k=3, debias_config=debias_config),
+            "debias_accuracy@3": Accuracy(k=3, debias_config=debias_config),
+            "debias_mcc@3": MCC(k=3, debias_config=debias_config),
+            "debias_hitrate@3": HitRate(k=3, debias_config=debias_config),
+            "debias_map@1": MAP(k=1, debias_config=debias_config),
+            "debias_map@3": MAP(k=3, debias_config=debias_config),
+            "debias_ndcg@3": NDCG(k=3, debias_config=debias_config),
+            "debias_mrr@3": MRR(k=3, debias_config=debias_config),
+            "debias_pap@3": PAP(k=3, debias_config=debias_config),
+            "debias_partauc@3": PartialAUC(k=3, debias_config=debias_config),
+        }
+        metrics = {
+            "debias_precision@3": Precision(k=3),
+            "debias_rprecision@3": Precision(k=3, r_precision=True),
+            "debias_recall@3": Recall(k=3),
+            "debias_f1beta@3": F1Beta(k=3),
+            "debias_accuracy@3": Accuracy(k=3),
+            "debias_mcc@3": MCC(k=3),
+            "debias_hitrate@3": HitRate(k=3),
+            "debias_map@1": MAP(k=1),
+            "debias_map@3": MAP(k=3),
+            "debias_ndcg@3": NDCG(k=3),
+            "debias_mrr@3": MRR(k=3),
+            "debias_pap@3": PAP(k=3),
+            "debias_partauc@3": PartialAUC(k=3),
+        }
+
+        interactions_downsampling = debias_interactions(self.interactions, config=debias_config)
+
+        actual = calc_metrics(
+            metrics=debias_metrics,  # type: ignore
```

Comment:
Why do we need ignore here and below?

Reply:
I use this 'ignore' because i found an interesting situation, the lint gives an error when i use only metric as DebiasableMetrikAtK in dict, but i dont  give the dictionary metrics that are directly inherited by MetrikAtK

Reply:
Could you elaborate a bit on this? I don't understand

Reply:
If I remove `# type: ignore`, I get the following error from `mypy` linter: 
![5298987039439576538](https://github.com/user-attachments/assets/acf1bd4a-58a4-4af9-bf93-a279acf072aa)

But if I add to the current metrics dictionaries in this test that don't inherit from `DebiasableMetricAtK`, there is no error.

And since this test is aimed at testing `DebiasableMetricAtK` metrics, it won't be correct to add here metrics that don't belong to this class.

If you know how to solve this problem, I would be grateful.

Reply:
Ok, for the future: we're using linters to make the code better and avoid as many mistakes as possible. If linter gives us an error, we study it and try to fix it. If you have a problem with understanding a problem feel free to ask, we are happy to help. We ignore linter only when we are absolutely sure that the error is false positive. 
If we ignored every error we didn't understand, then it wouldn't make any sense to use linters, right?

If mypy gives you a link to a page with the problem explanation, it could be very useful to read it sometimes :)  
Here to fix the problem it's enough to add an explicit type annotation like `metrics: tp.Dict[str, MetricAtK]`, but it's better to use the hint from mypy and change `Dict` to `Mapping` in the function type annotation

Reply:
Is it okay if I change the annotation in the [select_by_type](https://github.com/In48semenov/RecTools/blob/feature/debias_wrapper/rectools/metrics/scoring.py#L120) function for `objects` from `Dict` to `Mapping` along with this change? 
Because after changing the annotation in `calc_metrics` mypy gives the following error:
```
Argument 1 to "select_by_type" has incompatible type "Mapping[str, MetricAtK]"; expected "Dict[Any, Any]"
```

Reply:
Yes, you can change it to `Mapping[Any, Any]`  
Let me know if you get any new error from mypy